lib/src/room.dart

```diff
@@ -148,6 +158,32 @@ class KurentoRoom extends ChangeNotifier implements Room {
     // HACK: The `_serviceRequestPresenceTopic` has been unreliable and we haven't yet figured out why. Until then,
     // we're backing it up by periodically checking the status of the service request.
     _getServiceRequestStatusTimer = Timer.periodic(const Duration(seconds: 3), (_) => _getServiceRequestStatus());
+
+    if (defaultTargetPlatform == TargetPlatform.iOS || defaultTargetPlatform == TargetPlatform.android) {
+      _connectivityMonitoringSubscription = Connectivity().onConnectivityChanged.listen(_handleConnectionRecovery);
+    }
+  }
+
+  Future<void> _handleConnectionRecovery(ConnectivityResult result) async {
+    if (result != _currentlyUsedConnectionType) { // If status changed
+      if (ServiceRequestState.started == _serviceRequestState) { // If communication was previously established
+        if (ConnectivityResult.none == result) {
+          _log.info('Lost connection to Internet');
+          onConnectionLost?.call();
+        } else {
+          _log.info('Now using connection type: $result. '
+              'Restarting ice on ${_connectionByTrackId.length} webrtc connections!');
+          for (var element in _connectionByTrackId.values) {
+            await element.restartIce();
+          }
+          if (ConnectivityResult.none == _currentlyUsedConnectionType) {
+            await _reconnect();
+          }
```

Comment:
when the connection was lost (nothing to replace) and then retrieved, it seems to work better when we follow with a reconnect.